filament/backend/src/metal/PlatformMetal.mm

```diff
@@ -48,7 +49,29 @@
     return MetalDriverFactory::create(this, driverConfig);
 }
 
+
+bool PlatformMetal::initialize() noexcept {
+    MetalDevice device{};
+    createDevice(device);
+    if (device.device == nil) {
+        return false;
+    }
+
+    MetalCommandQueue commandQueue{};
+    createCommandQueue(device, commandQueue);
+    if (commandQueue.commandQueue == nil) {
+        return false;
```

Comment:
wouldn't we leak the MetalDevice here?

Reply:
No, the device is stored inside `pImpl->mDevice`. It will be released when `PlatformMetal` is destroyed.